src/components/reusables/Filter/Filter.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import { useState, useEffect, useId, type ReactNode } from 'react';
+import { Button } from '@digdir/design-system-react';
+import cn from 'classnames';
+import { XMarkIcon } from '@navikt/aksel-icons';
+import { useTranslation } from 'react-i18next';
+
+import { arraysEqual } from '@/resources/utils';
+import { usePrevious } from '@/resources/hooks';
+
+import type { FilterOption } from './utils';
+import { OptionDisplay } from './OptionDisplay';
+import classes from './Filter.module.css';
+import { Floatover } from './Floatover';
+import { FilterButton } from './FilterButton';
+
+export interface FilterProps {
+  options: FilterOption[];
+  label: string;
+  applyButtonLabel: string;
+  resetButtonLabel: string;
+  searchable?: boolean;
+  icon?: ReactNode;
+  value?: string[];
+  onApply?: (value: string[]) => void;
+  modalView?: boolean;
+  closeButtonAriaLabel?: string;
+}
+
+export const Filter = ({
+  options,
+  label,
+  applyButtonLabel,
+  resetButtonLabel,
+  icon,
+  value,
+  searchable,
+  modalView = false,
+  closeButtonAriaLabel,
+  onApply,
+}: FilterProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [activeFilters, setActiveFilters] = useState<string[]>(value ?? []);
+  const [checkedFilters, setCheckedFilters] = useState<string[]>(value ?? []);
+  const [hasChanges, setHasChanges] = useState(false);
+  const filterButtonID = useId();
+  const { t } = useTranslation('common');
+
+  // Update selected values when there are external changes
+  const prevValue = usePrevious(value);
+  useEffect(() => {
+    if (value !== undefined && !arraysEqual(value, prevValue)) {
+      setActiveFilters(value);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    setHasChanges(!arraysEqual(activeFilters, checkedFilters));
+  }, [checkedFilters]);
+
+  const handleApply = () => {
+    onApply?.(checkedFilters);
+    setActiveFilters([...checkedFilters]);
+    setHasChanges(false);
+  };
+
+  const handleReset = () => {
+    setCheckedFilters([]);
+  };
+
+  const handleOpenOrClose = () => {
+    if (isOpen) {
+      handleApply();
+      setIsOpen(false);
+    } else {
+      setIsOpen(true);
+    }
+  };
+
+  const modalHeader = () => (
+    <div className={classes.modalHeader}>
+      <h3>{label}</h3>
+      <Button
+        variant='quiet'
+        color='secondary'
+        onClick={handleOpenOrClose}
+        icon={<XMarkIcon />}
```

Comment:
We need alt-text on this icon

Reply:
Isn't the aria-label enough?

Reply:
Ahh, thought you could add alt-text as a prop but seems like you need to use aria-label isntead. Google says you can use aria-label instead so I assume it works
```suggestion
        icon={<XMarkIcon aria-label='closeIcon'/>}
```

Reply:
Thanks, then I'll do that instead 😊